argo_workflow_tools/dsl/dag_compiler.py

```diff
@@ -479,13 +481,14 @@ def _build_dag_template(node: DAGNode) -> argo.Template:
     return dag_template
 
 
-def compile_dag(entrypoint: DAGNode, on_exit: DAGNode = None) -> argo.WorkflowSpec:
+def compile_dag(entrypoint: DAGNode, on_exit: DAGNode = None, use_workflow_template_refs: bool = True) -> argo.WorkflowSpec:
```

Comment:
my comment - 
1. rename it to - embed_workflow_templates
2. invert it to be false by default.